### PR TITLE
fix: Backtrace has been removed from Error

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

This PR will bump the toolchain to `nightly-2022-08-11`, which is the same as the current stable (1.63) but in nightly.

From rust 1.63, `backtrace` has been removed from `Error` trait.

**Checklist**

- [x] Updated guide with pertinent info (may not always apply). 
- [x] Squash down commits to one or two logical commits which clearly describe the work you've done.
- [x] Unittest is a friend:)

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/openraft/540)

### CODE REVIEW

**Comment by Xuanwo** [comment]
I will squash the commit right before merging.

**Comment by Xuanwo** [comment]
BTW, reviewable seems done ):

**Comment by Xuanwo** [comment]
Cc @drmingdrmer @lichuang for reviewing.

File: openraft/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,4 @@
 #![doc = include_str!("../../README.md")]
-#![cfg_attr(feature = "bt", feature(backtrace))]
-#![cfg_attr(feature = "bench", feature(test))]
```
Comment (drmingdrmer) [inline openraft/src/lib.rs:3]: `feature(test)` is still required when `bench` is enabled.

It's my fault, the benchmark is not covered in CI.

File: openraft/Cargo.toml [hunk 2]
```diff
@@ -55,7 +55,7 @@ bench = []
 
 # Enable backtrace when generating an error.
 # Stable rust does not support backtrace.
-bt  = ["anyerror/backtrace", "anyhow/backtrace"]
+bt  = ["anyhow/backtrace"]
```
Comment (drmingdrmer) [inline openraft/Cargo.toml:58]: The backtrace feature in `anyerror` is still needed to be turned on.

**Review by drmingdrmer — COMMENTED** [review]
Nice shot!
Less codes less troubles :D

File: openraft/Cargo.toml (hunk 2 above)
Comment (Xuanwo) [reply openraft/Cargo.toml:58]: anyerror needs an upgrade to make it work on rust 1.63.

File: openraft/src/config/config_test.rs [hunk 3]
```diff
@@ -79,50 +79,50 @@ fn test_build() -> anyhow::Result<()> {
...
-    assert_eq!(true, config.enable_elect);
+    assert!(config.enable_elect);
 
     let config = Config::build(&["foo", "--enable-elect"])?;
-    assert_eq!(true, config.enable_elect);
+    assert!(config.enable_elect);
 
     let config = Config::build(&["foo"])?;
-    assert_eq!(true, config.enable_elect);
+    assert!(config.enable_elect);
```
[the remark this answers is not in the document]
Comment (Xuanwo) [reply openraft/src/config/config_test.rs:125]: If so, we can disable this check from clippy.

File: openraft/src/core/raft_core.rs [hunk 4]
```diff
@@ -797,6 +797,7 @@ impl<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> RaftCore<C,
 
     /// Get the next election timeout, generating a new value if not set.
     /// TODO: get() should not have a side effect of updating the timer.
+    #[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (Xuanwo) [reply openraft/src/core/raft_core.rs:800]: If this function is not used, we can remove it.

File: openraft/tests/benchmark/bench_cluster.rs [hunk 5]
```diff
@@ -68,8 +68,8 @@ fn bench_with_config(bench_config: &BenchConfig) -> anyhow::Result<()> {
         .build()?;
 
     // Run client_write benchmark
-    let output = rt.block_on(do_bench(bench_config))?;
-    Ok(output)
+    rt.block_on(do_bench(bench_config))?;
+    Ok(())
```
[the remark this answers is not in the document]
Comment (Xuanwo) [reply openraft/tests/benchmark/bench_cluster.rs:72]: Maybe we can disable this check from clippy.

File: openraft/tests/fixtures/mod.rs [hunk 6]
```diff
@@ -405,7 +405,10 @@ where
...
     /// Get a payload of the latest metrics from each node in the cluster.
     pub fn latest_metrics(&self) -> Vec<RaftMetrics<C::NodeId, C::Node>> {
-        let rt = self.routing_table.lock().unwrap();
+        // `watch::Receiver<RaftMetrics<T>>` holds another locks, so we should `clone` the map here
+        // to avoid `clippy::has_significant_drop`.
+        let rt = self.routing_table.lock().unwrap().clone();
+
         let mut metrics = vec![];
         for node in rt.values() {
             metrics.push(node.0.metrics().borrow().clone());
```
[the remark this answers is not in the document]
Comment (Xuanwo) [reply openraft/tests/fixtures/mod.rs:414]: `node.0.metrics().borrow()` holds a lock internally.

File: openraft/Cargo.toml (hunk 2 above)
Comment (drmingdrmer) [reply openraft/Cargo.toml:58]: I fired an updated release of anyerror to adapt changes introduced by stable-1.63 and the latest nightly:
https://crates.io/crates/anyerror/0.1.7

**Comment by drmingdrmer** [comment]
It involves some other modification to let the CI pass and be complete. Manually merged.

**Pull request merged**